src/MarginModels/MarginBase.py

```diff
@@ -0,0 +1,162 @@
+# Copyright 2020, Battelle Energy Alliance, LLC
+# ALL RIGHTS RESERVED
+"""
+Created on May 13 2021
+
+@author: mandd,wangc
+"""
+
+#External Modules------------------------------------------------------------------------------------
+import os
+import abc
+import sys
+import numpy as np
+#External Modules End--------------------------------------------------------------------------------
+
+#Internal Modules------------------------------------------------------------------------------------
+from utils import mathUtils as utils
+from utils import InputData, InputTypes
+#Internal Modules End-
+
+class MarginBase():
+  """
+    Base class for margin models
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Collects input specifications for this class.
+      @ In, cls, class instance
+      @ Out, inputSpecs, InputData, specs
+    """
+    inputSpecs = InputData.parameterInputFactory('MarginModel')
+    inputSpecs.addParam('type', param_type=InputTypes.StringType, descr='The margin model object identifier')
+    return inputSpecs
+
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    self.type = self.__class__.__name__
+    self.name = self.__class__.__name__
+    # dictionary: keys all required input parameters, and values either user provided values or
+    # variables determined by raven
+    self._variableDict = {}
+    # instance of margin model
+    self._model = None
+    # class of model
+    self._modelClass = None
+    # variable stores margin value
+    self._margin = None
+
+  def handleInput(self, xmlNode):
+    """
+      Function to read the portion of the xml input that belongs to this specialized class
+      and initialize some stuff based on the inputs got
+      @ In, xmlNode, xml.etree.ElementTree.Element, Xml element node
+      @ Out, None
+    """
+    paramInput = self.getInputSpecification()()
+    paramInput.parseNode(xmlNode)
+    self._localHandleInput(paramInput)
+
+  @abc.abstractmethod
+  def _localHandleInput(self, paramInput):
```

Comment:
In RAVEN, we are trying to use inheritance instead to create local method. I suggest we follow the same strategy for SR2ML